src/components/dashboard/clickable_cell_store.ts

```diff
@@ -0,0 +1,86 @@
+import { markRaw } from "@odoo/owl";
+import { positionToZone, toXC } from "../../helpers";
+import { clickableCellRegistry } from "../../registries/cell_clickable_registry";
+import { SpreadsheetStore } from "../../stores/spreadsheet_store";
+import {
+  CellPosition,
+  Command,
+  Rect,
+  SpreadsheetChildEnv,
+  UID,
+  Zone,
+  invalidateEvaluationCommands,
+} from "../../types";
+
+type Garbage = ((position: CellPosition, env: SpreadsheetChildEnv) => void) | false;
+
+export interface ClickableCell {
+  coordinates: Rect;
+  position: CellPosition;
+  action: (position: CellPosition, env: SpreadsheetChildEnv) => void;
+}
+
+export class ClickableCellsStore extends SpreadsheetStore {
+  private _clickableCells: Record<UID, Record<string, Garbage>> = markRaw({});
+
+  handle(cmd: Command) {
+    if (
+      invalidateEvaluationCommands.has(cmd.type) ||
+      cmd.type === "EVALUATE_CELLS" ||
+      (cmd.type === "UPDATE_CELL" && cmd.content)
+    ) {
+      this._clickableCells = markRaw({});
+    }
+  }
+
+  private getClickableAction(position: CellPosition): Garbage {
+    const { sheetId, col, row } = position;
+    const clickableCells = this._clickableCells;
+    const xc = toXC(col, row);
+    if (!clickableCells[sheetId]) {
+      clickableCells[sheetId] = {};
+    }
+    if (!(xc in clickableCells[sheetId]!)) {
+      clickableCells[sheetId][xc] = this.findClickableAction(position);
+    }
+    return clickableCells[sheetId][xc];
+  }
+
+  private findClickableAction(position: CellPosition) {
+    const getters = this.getters;
+    for (const item of clickableCellRegistry.getAll().sort((a, b) => a.sequence - b.sequence)) {
+      if (item.condition(position, getters)) {
+        return item.execute;
+      }
+    }
+    return false;
+  }
+
+  get clickableCells(): ClickableCell[] {
+    const cells: ClickableCell[] = [];
+    const getters = this.getters;
+    const sheetId = getters.getActiveSheetId();
+    for (const col of getters.getSheetViewVisibleCols()) {
+      for (const row of getters.getSheetViewVisibleRows()) {
+        const position = { sheetId, col, row };
+        const action = this.getClickableAction(position);
+        if (!action) {
+          continue;
+        }
+        let zone: Zone;
+        if (getters.isInMerge(position)) {
+          zone = getters.getMerge(position)!;
```

Comment:
In the case of emrges, we will re-compute  `this.getClickableAction` for each cell of the merge, and have a ClickableCell with the same zone for each cell right ? That doesn't seems right ... (we love merges)

Reply:
agreed that it's a waste of resource but not fundamentally wrong ^^

Reply:
it could be fixed in prior versions as well actually